### DISCUSSION TITLE
ESP32-S3 ST7701 No mem for frame buffer?

### DISCUSSION THREAD

**Discussion by showdown** [moononournation/Arduino_GFX#546] [General]
Arduino IDE 2.3.3
ESP board manager 2.0.17
Arduino_GFX
LittleFS

Waveshare ESP32-S3-LCD: ESP32-S3-WROOM-1-N8R8
2.8" ST7701 round display

I can get the helloworld, clock, wifi analyzer, and ASCII table to compile, load, run on the board and display but when I try to use the Sprite or imageViewer, after successful upload with LittleFS I receive this error: 

`E (247) lcd_panel.rgb: esp_lcd_new_rgb_panel(151): no mem for frame buffer
ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4037ab90
file: "/Users/julianbaumgartner/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_ESP32RGBPanel.cpp" line 118
func: uint16_t* Arduino_ESP32RGBPanel::getFrameBuffer(int16_t, int16_t)
expression: esp_lcd_new_rgb_panel(_panel_config, &_panel_handle)

abort() was called at PC 0x4037ab93 on core 1

Backtrace: 0x4037777e:0x3fceba90 0x4037ab9d:0x3fcebab0 0x403807f5:0x3fcebad0 0x4037ab93:0x3fcebb50 0x42003e33:0x3fcebb70 0x420044a0:0x3fcebba0 0x42002002:0x3fcebbc0 0x420099cb:0x3fcebc50
`

Tried every partition with SPIFFS with no change in the error.

I haven't the slightest clue why this is happening.

**Comment by zz6zz666** [comment]
我在使用基于esp_idf的乐鑫官方st7701驱动库的时候，遇到了相同的问题

**Comment by zz6zz666** [comment]
解决方案是，启用分配在psram选项（esp_idf）
![image](https://github.com/user-attachments/assets/c0571bf4-c027-458b-8107-b4ed95eaba6e)
